### PR TITLE
Charts improvements and bug fixes

### LINKED ISSUE #175: Charts improvements and bug fixes
PR changes:
- NPE fixes
- added JVM remote debug flags
- wget cert validation under flag for VPN issues (blocked in internal N/W)
- added values file for local setup

TODO:
- [ ] cleanup
- [ ] readme update
- [ ] remote args under flag

### PR DESCRIPTION
**Pull request by iosdev747**
PR changes:
- NPE fixes
- added JVM remote debug flags
- wget cert validation under flag for VPN issues (blocked in internal N/W)
- added values file for local setup

TODO:
- [ ] cleanup
- [ ] readme update
- [ ] remote args under flag

### CODE REVIEW

File: setup/helm/varadhi/values.yaml [hunk 1]
```diff
@@ -212,12 +219,12 @@ messaging:
   configMapName: messaging
   pulsar:
     adminOptions:
-      serviceHttpUrl: "http://192.168.1.3:8080"
+      serviceHttpUrl: "http://host.docker.internal:8080"
```
Comment (kmrdhruv) [inline setup/helm/varadhi/values.yaml:222]: may be time to decide the hierarchy of values files. Suggestion/TBD : values -> fcp.values ->  role specific i.e (controller|consumer|server|standalone) and override only required content in higher level files ?

File: setup/docker/scripts/entrypoint.sh [hunk 2]
```diff
@@ -16,4 +16,11 @@ JAVA_OPTS="$JAVA_OPTS -Dcom.sun.management.jmxremote.authenticate=false"
 JAVA_OPTS="$JAVA_OPTS -Dcom.sun.management.jmxremote.ssl=false"
 JAVA_OPTS="$JAVA_OPTS -Dlog4j2.configurationFile=/etc/varadhi/log4j2.xml"
 
+# If debug flag is set(1 or true) then enable remote debugging
+if [[ ! -z "$APP_DEBUG" ]]; then
+    if [[ "$APP_DEBUG" == "1" || "$APP_DEBUG" == "true" ]]; then
```
Comment (kmrdhruv) [inline setup/docker/scripts/entrypoint.sh:21]: any issue being it just set to any value ?

**Review by kmrdhruv — CHANGES_REQUESTED** [review]

File: server/src/main/java/com/flipkart/varadhi/config/MemberConfig.java [hunk 3]
```diff
@@ -9,4 +9,6 @@ public class MemberConfig {
     private int clusterPort;
     private int maxQps;
     private int networkMBps;
+    private int cpuCount;
+    private int nicMBps;
 }
```
Comment (kmrdhruv) [inline server/src/main/java/com/flipkart/varadhi/config/MemberConfig.java:14]: instead we can adjust the value/config.

File: setup/docker/Dockerfile [hunk 4]
```diff
@@ -19,14 +22,20 @@ RUN setcap cap_net_raw,cap_net_admin=eip /usr/bin/ngrep
 # Install Zookeeper client.
 RUN mkdir -p /tools/zk
 WORKDIR /tools/zk
-RUN wget https://archive.apache.org/dist/zookeeper/zookeeper-3.6.0/apache-zookeeper-3.6.0-bin.tar.gz
+RUN if "$SKIP_CERT_CHECK";  \
+    then wget https://archive.apache.org/dist/zookeeper/zookeeper-3.6.0/apache-zookeeper-3.6.0-bin.tar.gz --no-check-certificate ;  \
```
[the remark this answers is not in the document]
Comment (iosdev747) [reply setup/docker/Dockerfile:25]: No change are required on github runner.
This is for local (under VPN). We can use `--build-arg SKIP_CERT_CHECK=true` while building the docker image.

File: setup/docker/Dockerfile [hunk 5]
```diff
@@ -66,5 +75,7 @@ RUN sed -i 's|src/main/resources/|/etc/varadhi/|' /etc/varadhi/configuration.yml
 USER varadhi
 WORKDIR /usr/share/varadhi
 
+RUN env
```
[the remark this answers is not in the document]
Comment (iosdev747) [reply setup/docker/Dockerfile]: removed it, accidentally added it

File: setup/helm/test/service-server.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+apiVersion: v1
```
[the remark this answers is not in the document]
Comment (iosdev747) [reply setup/helm/test/service-server.yaml:1]: no, I left it here for my reference. Will remove it.

File: setup/docker/scripts/entrypoint.sh (hunk 2 above)
Comment (iosdev747) [reply setup/docker/scripts/entrypoint.sh:21]: no, idea was if someone sets to false or 0 or any other invalid value, we don't enable it.

File: setup/helm/varadhi/values/controller_values.yaml [hunk 7]
```diff
@@ -0,0 +1,240 @@
+# Default values for varadhi.
```
Comment (kmrdhruv) [inline setup/helm/varadhi/values/controller_values.yaml:1]: This is duplicating the entire values files for each component, that shouldn't be needed. What is the proposed hierarchy ?

**Review by kmrdhruv — CHANGES_REQUESTED** [review]

File: server/src/main/java/com/flipkart/varadhi/config/MemberConfig.java [hunk 8]
```diff
@@ -9,4 +9,6 @@ public class MemberConfig {
     private int clusterPort;
     private int maxQps;
     private int networkMBps;
+    private int cpuCount;
+    private int nicMBps;
```
Comment (gauravAshok) [inline server/src/main/java/com/flipkart/varadhi/config/MemberConfig.java:13]: nicMBps and networkMBps both mean the same thing.

File: setup/docker/scripts/entrypoint.sh (hunk 2 above)
Comment (gauravAshok) [reply setup/docker/scripts/entrypoint.sh:21]: Yup. only inner check is enough

File: setup/docker/scripts/entrypoint.sh [hunk 9]
```diff
@@ -16,4 +16,11 @@ JAVA_OPTS="$JAVA_OPTS -Dcom.sun.management.jmxremote.authenticate=false"
...
 
-exec java -cp ./*:dependencies/* $JAVA_OPTS com.flipkart.varadhi.VaradhiApplication /etc/varadhi/configuration.yml
+# If debug flag is set(1 or true) then enable remote debugging
+if [[ ! -z "$APP_DEBUG" ]]; then
+    if [[ "$APP_DEBUG" == "1" || "$APP_DEBUG" == "true" ]]; then
+        JAVA_OPTS="$JAVA_OPTS -agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=*:5005"
+    fi
+fi
+
+exec java -cp ./*:dependencies/* $JAVA_OPTS $JAVA_EXTRA_OPTS com.flipkart.varadhi.VaradhiApplication /etc/varadhi/configuration.yml
```
Comment (gauravAshok) [inline setup/docker/scripts/entrypoint.sh:26]: What is JAVA_EXTRA_OPTS?

File: setup/helm/varadhi/values/controller_values.yaml [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+    syncLimit: 15
+    skipACL: yes
+    purgeInterval: 24
+    snapRetainCount: 10
+    peerType: "participant"
+
+images:
+  pullPolicy: IfNotPresent
+  appImage:
+    registry: jfrog.fkinternal.com
```
Comment (gauravAshok) [inline setup/helm/varadhi/values/controller_values.yaml:123]: this should not be present here.

File: setup/helm/varadhi/values/controller_values.yaml [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+  configMapName: logging
+  level: info
+
+#############################################
+# Metastore config -- zookeeper
+#############################################
+metastore:
+  configMapName: metastore
+  zookeeper:
+    connectUrl: "127.0.0.1:2181"
```
Comment (gauravAshok) [inline setup/helm/varadhi/values/controller_values.yaml:203]: in server value file this is set as 
varadhi-zookeeper-0.varadhi-zookeeper.default.svc.cluster.local. why?

**Comment by iosdev747** [comment]
Continued at https://github.com/flipkart-incubator/varadhi/pull/190

**Pull request closed without merging**